preview/Win7Msix/Win7MSIXInstaller/ValidateTargetDeviceFamily.cpp

```diff
@@ -0,0 +1,118 @@
+#include <windows.h>
+
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+
+#include "FilePaths.hpp"
+#include "ValidateTargetDeviceFamily.hpp"
+#include "GeneralUtil.hpp"
+#include <TraceLoggingProvider.h>
+#include <VersionHelpers.h>
+
+const PCWSTR ValidateTargetDeviceFamily::HandlerName = L"ValidateRequest";
+
+HRESULT ValidateTargetDeviceFamily::ExecuteForAddRequest()
+{
+    RETURN_IF_FAILED(ParseTargetDeviceFamilyFromPackage());
+    if ((m_targetDeviceFamilyName == L"MSIXCore.Desktop" || m_targetDeviceFamilyName == L"MSIXCore.Server")

```

Comment:
>m_targetDeviceFamilyName == L"MSIXCore.Desktop" || m_targetDeviceFamilyName == L"MSIXCore.Server") [](start = 9, length = 98)

i think this needs to be a separate function that determines whether the targetDeviceFamilyName matches the current OS. Server OS should accept both, but Desktop OS should only accept MSIXCore.Desktop.

Reply:
Added method to compare targetDeviceFamilyName with Desktop or Server OS. Added methods isWindowsProductTypeDesktop and isWindowsProductTypeServer to identify between desktop or server OS. This is called while comparing targetdevicefamilyname and version check.